api/v1/installation_types.go

```diff
@@ -178,6 +179,35 @@ type InstallationSpec struct {
 	// +kubebuilder:validation:Enum=Enabled;Disabled
 	// +optional
 	FIPSMode *FIPSMode `json:"fipsMode,omitempty"`
+
+	// Logging Configuration for Components
+	// +optional
+	Logging *Logging `json:"logging,omitempty"`
+}
+
+type Logging struct {
+	// Customized logging specification for calico-cni plugin
+	// +optional
+	CNILogging *CNILogging `json:"cniLogging,omitempty"`
+}
+
+type CNILogging struct {
+	// Default: Info
+	// +kubebuilder:validation:Enum=Error;Warning;Debug;Info
+	// +optional
+	LogSeverity *LogLevel `json:"logSeverity,omitempty"`
+
+	// Default: 100Mi
```

Comment:
In order for the default values set by kubebuilder, I need to remove the code in core_controller (fillDefaults). O.w., the values set in fillDefaults will take place.
Looks like environment is set differently for some tests and default values are not set via kubebuilder there. Thus, the side effect of not setting default values in fillDefaults is that some tests are failing and we need to check for nil values while we are setting configmap (node.go). This means that for some of tests, cni-config will not have the logging properties. This is a different behaviour from production, where defaults are always set via kubebuilder and therefore the logging values will always be present in cni-config (whether user provides it or not).

What should we do in this case?
I am more towards using fillDefaults over kubebuilder since this is how default values for most of the properties in the installation object is defined already and tests are build around this setup.

Reply:
note: when we implement defaulting both in fillDefaults and kubebuilder, the ones in fillDefault are what we see and values set in kubebuilder:default tag are being ignored (I am not sure what is happening behind the scene - this is just observations from testing various scenarios)